src/components/MovieCard.js

```diff
@@ -1,19 +1,71 @@
 import "./MovieCard.css";
-import React from "react";
+import React, {useEffect, useRef, useState, Suspense} from "react";
 import PropTypes from "prop-types";
 
-const  MovieCard = ({movie}) => (
-    <div className="movie-card">
-        <img src={require(`../static/images/movies/${movie.icon}`)}/>
-        <div className="movie-card-info">
-            <div className="movie-card-name">{movie.name}</div>
-            <div className="movie-card-genres">
-                {movie.genres.map(genre=>genre.name).join(", ")}
+const ContextMenu = React.lazy(()=> import("./ContextMenu"));
+
+
+const  MovieCard = function(props){
+    const [anchorPoint, setAnchorPoint] = useState({ x: 0, y: 0 });
+    const [showContextMenu, setShowContextMenu] = useState(false);
+
+    const closeContextMenu = ()=>{
+        setShowContextMenu(false)
+    }
+
+    const movie = props.movie;
+    if(movie === undefined){
+        movie = {};
+    }
+    if(movie.name === undefined){
+        movie.name = "Best Movie";
+    }
+    if(movie.year === undefined){
+        movie.year = 2000;
+    }
+    if(movie.icon === undefined){
+        movie.icon = "default.png";
+    }
+    if(movie.genres === undefined){
+        movie.genres = {}
+    }
```

Comment:
it's a little bit hardcoded, you could try to add it into [`defaultProps`](https://ru.reactjs.org/docs/typechecking-with-proptypes.html)

Reply:
how do I use default props on nested object field level?